File: HackerRank/Algorithms/1. Warmup/Diagonal_difference.cpp
#include <cmath>
#include <vector>
#include <iostream>
using namespace std;

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    ios::sync_with_stdio(false);
    int N;
    cin >> N;
    vector< vector<int> > matrix(N, vector<int>(N));
    int i, j, left = 0, right = 0;
    for (i = 0; i < N; ++i) {
        for(j = 0; j < N; ++j) {
            cin >> matrix[i][j];
        }
    }
    i = 0;
    while (i < N) {
        left += matrix[i][i];
        right += matrix[i][N-1-i];
        ++i;
    }
    cout << abs(left - right) << endl;
    return 0;
}

File: HackerRank/Algorithms/1. Warmup/Diagonal_difference.java
import java.util.Scanner;
import java.lang.Math;

public class Solution {

    public static int calculateDiagonalDifference(int[][] m, int n) {
        int left = 0, right = 0;
        int i = 0;
        while (i < n) {
            left += m[i][i];
            right += m[i][n-i-1];
            ++i;
        }
        return Math.abs(left-right);
    }

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();
        int[][] matrix = new int[N][N];
        for(int i = 0; i < N; ++i) {
            for(int j = 0; j < N; ++j) {
                matrix[i][j] = scan.nextInt();
            }
        }
        scan.close();
        System.out.println(calculateDiagonalDifference(matrix, N));
    }
}
